src/main/java/car/app/services/CarService.java

```diff
@@ -1,52 +1,81 @@
 package car.app.services;
 
-import java.sql.Timestamp;
-import java.util.ArrayList;
 import java.util.List;
-import java.util.Map;
 
 import javax.ejb.Stateless;
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import javax.validation.Valid;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
 
-import car.app.database.CarDatabase;
 import car.app.entity.Car;
+import car.app.entity.ErrorMessage;
 
 @Stateless
 public class CarService {
 
-	private Map<Long, Car> Cars = CarDatabase.getCars();
-
-	public CarService() {
-		Timestamp tm = new Timestamp(0);
-		Cars.put(1L, new Car(1, "coche1", tm, "España", tm, tm));
-		Cars.put(2L, new Car(2, "coche5", tm, "Alemania", tm, tm));
-	}
+	@PersistenceContext(unitName = "postg")
+	EntityManager em;
 
 	public List<Car> getCars() {
-		return new ArrayList<>(Cars.values());
+		String q = "from Car";
+		return em.createQuery(q, Car.class).getResultList();
 	}
 
-	public Car getCar(long CarId) {
-		return Cars.get(CarId);
+	public Car getCar(int id) {
+
+		ErrorMessage errorMessage = new ErrorMessage("CAR WITH ID " + id + " NOT FOUND", 404, "");
+		Response response = Response.status(Status.NOT_FOUND).entity(errorMessage).build();
+
+		Car car = em.find(Car.class, id);
+
+		if (car == null) {
+			throw new WebApplicationException(response);
+		}
+		return car;
 
 	}
 
-	public Car addCar(Car car) {
-		car.setId(Cars.size() + 1);
-		Cars.put((long) car.getId(), car);
+	public Car addCar(@Valid Car car) {
+		em.persist(car);
 		return car;
 	}
 
-	public Car updateCar(Car car) {
+	public Car updateCar(@Valid Car car) {
 		int id = car.getId();
-		if (id == 0) {
-			return null;
+		ErrorMessage errorMessage = new ErrorMessage("CAR WITH ID " + id + " NOT FOUND", 404, "");
+		Response response = Response.status(Status.NOT_FOUND).entity(errorMessage).build();
```

Comment:
This is already done by DataNotFoundExceptionMapper.java, so here should only trigger a DataNotFoundException passing the error in case the car is not there